code/game/objects/items/weapons/teleportation.dm

```diff
@@ -208,6 +186,38 @@ Frequency:
 		recharging = 1
 		processing_objects.Add(src)
 
+/obj/item/weapon/hand_tele/proc/choose_destination(var/user)
+	var/list/L = list(  )
+	for(var/obj/machinery/computer/teleporter/R in machines)
+		for(var/obj/machinery/teleport/hub/com in locate(R.x + 2, R.y, R.z))
+			if(R.locked && !R.one_time_use)
+				if(com.engaged)
+					L["[R.id] (Active)"] = R.locked
+				else
+					L["[R.id] (Inactive)"] = R.locked
+
+	var/list/turfs = new/list()
+
+	for (var/turf/T in trange(10, user))
+		// putting them at the edge is dumb
+		if (T.x > world.maxx - 8 || T.x < 8)
+			continue
+
+		if (T.y > world.maxy - 8 || T.y < 8)
+			continue
+
+		turfs += T
+
+	if (turfs.len)
+		L["None (Dangerous)"] = pick(turfs)
+
+	turfs = null
+	destination_name = input(user, "Please select a teleporter to lock in on.", "Hand Teleporter") in L
+	destination_id = L[destination_name]
+
+/obj/item/weapon/hand_tele/AltClick(var/mob/usr)
```

Comment:
`AltClick` doesn't have any sanity check, the only requirement for this to run is that someone alt-clicked the hand tele. That means ghosts can freely set the destination, for instance. Or people from across the room.
Add sanity checks.

Reply:
Also, call the parameter `user`, not `usr`